src/btc-faucet.ts

```diff
@@ -1,87 +1,159 @@
 import * as btc from 'bitcoinjs-lib';
-import * as BN from 'bn.js';
-
-type UnspentTxInfo = {
-  txid: string;
-  vout: number;
-  address: string;
-  acount: string;
-  scriptPubKey: string;
-  amount: number;
-  confirmations: number;
-  redeemScript: string;
-  spendable: boolean;
-  solvable: boolean;
-  safe: boolean;
-};
-
-export const makeBTCFaucetPayment = async (address: string) => {
-  const { BTC_FAUCET_PK, BTC_RPC_PORT, BTC_RPC_HOST, BTC_RPC_PW, BTC_RPC_USER } = process.env;
-  if (!BTC_FAUCET_PK || !BTC_RPC_PORT || !BTC_RPC_HOST || !BTC_RPC_PW || !BTC_RPC_USER) {
+import * as Bluebird from 'bluebird';
+import { parsePort } from './helpers';
+import coinSelect = require('coinselect');
+
+export function getFaucetPk(): string {
+  const { BTC_FAUCET_PK } = process.env;
+  if (!BTC_FAUCET_PK) {
+    throw new Error('BTC Faucet not fully configured.');
+  }
+  return BTC_FAUCET_PK;
+}
+
+export function getFaucetWallet(
+  network: btc.Network
+): { key: btc.ECPairInterface; wif: string; address: string } {
+  const wif = getFaucetPk();
+  const key = btc.ECPair.fromWIF(wif, network);
+  const addr = getKeyAddress(key);
+  return {
+    key: key,
+    wif: wif,
+    address: addr,
+  };
+}
+
+export function getKeyAddress(key: btc.ECPairInterface): string {
+  const { address } = btc.payments.p2pkh({
+    pubkey: key.publicKey,
+    network: key.network,
+  });
+  if (!address) {
+    throw new Error('address generation failed');
+  }
+  return address;
+}
+
+export function getRpcClient(): RPCClient {
+  const { BTC_RPC_PORT, BTC_RPC_HOST, BTC_RPC_PW, BTC_RPC_USER } = process.env;
+  if (!BTC_RPC_PORT || !BTC_RPC_HOST || !BTC_RPC_PW || !BTC_RPC_USER) {
     throw new Error('BTC Faucet not fully configured.');
   }
   const client = new RPCClient({
     url: BTC_RPC_HOST,
-    port: parseInt(BTC_RPC_PORT, 10),
+    port: parsePort(BTC_RPC_PORT),
     user: BTC_RPC_USER,
     pass: BTC_RPC_PW,
   });
+  return client;
+}
+
+interface TxOutSet {
+  bestblock: string;
+  height: number;
+  success: boolean;
+  total_amount: number;
+  txouts: number;
+  unspents: {
+    amount: number;
+    desc: string;
+    height: number;
+    scriptPubKey: string;
+    txid: string;
+    vout: number;
+  }[];
+}
+
+// Replace with client.estimatesmartfee() for testnet/mainnet
+const REGTEST_FEE_RATE = 2000;
+
+const MIN_TX_CONFIRMATIONS = 100;
+
+export async function makeBTCFaucetPayment(
+  network: btc.Network,
+  address: string,
+  /** Amount to send in BTC */
+  faucetAmount: number
+): Promise<{ txId: string; rawTx: string }> {
+  const client = getRpcClient();
+  const faucetWallet = getFaucetWallet(network);
 
-  const network = btc.networks.regtest;
+  const faucetAmountSats = faucetAmount * 1e8;
 
-  const faucetPK = btc.ECPair.fromWIF(BTC_FAUCET_PK, btc.networks.regtest);
-  const faucetPayment = btc.payments.p2wpkh({
-    pubkey: faucetPK.publicKey,
-    network,
+  const txOutSet: TxOutSet = await client.scantxoutset({
```

Comment:
qq - why are we using `scantxoutset` (vs `listunspent` - not flagged with an experimental warning in the Core API)?

Reply:
`listunspent` is a wallet function. If we are going to use wallet functions then the whole faucet is maybe 3 lines of code that call `sendtoaddress`. My understanding is that we want the faucet to be able to operate given a keypair with some regtest btc. 

Also `scantxoutset` has been around for 2-3 years.

Reply:
@ludo let me know if the wallet RPC functions are preferable. I'm going to merge this for now so devops can get a deployment working

Reply:
stacks-blockchain is already using some wallet functions namely "listunspent" and "importaddress". I was also under the impression that the function `sendtoaddress` would be use, but don't have a strong opinion, and can add this additional call.